crypto/provider_core.c

```diff
@@ -1694,7 +1694,9 @@ static int core_obj_add_sigid(const OSSL_CORE_HANDLE *prov,
 static int core_obj_create(const OSSL_CORE_HANDLE *prov, const char *oid,
                            const char *sn, const char *ln)
 {
-    return (OBJ_create(oid, sn, ln) != NID_undef);
+    /* Create it, or verify it already exists if we fail to create */
+    return (OBJ_create(oid, sn, ln) != NID_undef
```

Comment:
Can we please drop the outermost parenthesis?

Reply:
I agree.  Besides, we get a fairly beautiful alignment without them:

``` C
    return OBJ_create(oid, sn, ln) != NID_undef
        || OBJ_txt2nid(oid) != NID_undef)
```

Reply:
Should the order be OBJ_txt2nid(oid) first and then OBJ_create if that failed? I.E., to prefer the read-only operation.

Reply:
Fixed, and order modified as per @t8m's suggestion.